subtrack.MAUI/Services/AutoPaymentHandler.cs

```diff
@@ -0,0 +1,45 @@
+﻿using subtrack.DAL.Entities;
+using subtrack.MAUI.Services.Abstractions;
+using subtrack.MAUI.Utilities;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace subtrack.MAUI.Services
+{
+    public class AutoPaymentHandler
+    {
+        private ISubscriptionService _subscriptionService;
+        private IDateTimeProvider _dateTimeProvider;
+        private ISettingsService _settingsService;
+        private GetSubscriptionsFilter filter;
+        
+        private bool HasExecutedToday(Subscription subscription)
```

Comment:
This should simply make use of the ISettingsService to check if the last time ExecuteAsync() was called today so it should not take in any subscription.